File: kattis/Backspace/Backspace.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.List;
import java.util.Stack;

public class Backspace {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String line = sc.nextLine();
        char[] chars = line.toCharArray();
        Stack<Character> stack = new Stack<Character>();
        for (int i = chars.length-1; i >= 0; i--) {
            stack.push(line.charAt(i));
        }
        StringBuilder sb = new StringBuilder();
        while (!stack.empty()) {
            char c = stack.pop();
            if (c == '<') {
                if (sb.length() > 0) {
                    sb.deleteCharAt(sb.length() - 1);
                }
            } else {
                sb.append(c);
            }
        }
        if (sb.length() > 0) {
            System.out.println(sb.toString());
        }
    }
}

File: kattis/Backspace/backspace.cpp
#include <stack>
#include <string>
#include<bits/stdc++.h>
using namespace std;


void showstack(stack<char> s)
{
    string str = "";
    stack<char> stek;
    while (!s.empty())
    {
        str += s.top();
        s.pop();
    }
    reverse(str.begin(), str.end());
    cout << str;
}

int main()
{
    stack<char> stek;
    string s;
    char c;
    cin >> s;
    for (int i = 0; i < s.length(); i++)
    {
        if (s[i] == '<')
        {
            stek.pop();
        }
        else
        {
            stek.push(s[i]);
        }
    }
    showstack(stek);
}
